site-content/source/modules/ROOT/pages/development/dependencies.adoc

```diff
@@ -2,50 +2,83 @@
-pay attention to any emerging issues in the `ci` and reported related
-issues on Jira/ML, in case of any project dependency changes.
+New dependencies should not be included without community consensus first being
+obtained via a `[DISCUSS]` thread on the dev@cassandra.apache.org mailing list.
 
 As Cassandra is an Apache product, all included libraries must follow
 Apache's https://www.apache.org/legal/resolved.html[software license
 requirements].
 
-=== Required steps to add or update libraries
-
-* Add or replace jar file in `lib` directory
-* Add or update `lib/license` files
-* Update dependencies in `build.xml`
-** Add to `parent-pom` with correct version
-** Add to `all-pom` if simple Cassandra dependency (see below)
-
-=== POM file types
-
-* *parent-pom* - contains all dependencies with the respective version.
-All other poms will refer to the artifacts with specified versions
-listed here.
-* *build-deps-pom(-sources)* + *coverage-deps-pom* - used by `ant build`
-compile target. Listed dependenices will be resolved and copied to
-`build/lib/{jar,sources}` by executing the
-`maven-ant-tasks-retrieve-build` target. This should contain libraries
-that are required for build tools (grammar, docs, instrumentation), but
-are not shipped as part of the Cassandra distribution.
-* *test-deps-pom* - refered by `maven-ant-tasks-retrieve-test` to
-retrieve and save dependencies to `build/test/lib`. Exclusively used
-during JUnit test execution.
-* *all-pom* - pom for
-https://mvnrepository.com/artifact/org.apache.cassandra/cassandra-all[cassandra-all.jar]
-that can be installed or deployed to public maven repos via
-`ant publish`
+Cassandra uses the Ant build system and Maven POMs for dependency specification.
+In Cassandra 4.2 the format of POMs was moved from within the `build.xml` file to separate POM template files that are processed by Ant.
+In both pre-4.2 and post-4.2 Cassandra, there are several POMs that dependencies can be included in:
+
+* *parent-pom*
+    - Contains all dependencies with the respective version. All other poms
+      will refer to the artifacts with specified versions listed here.
+    - Since Cassandra 4.2, the `parent-pom` template is `.build/parent-pom-template.xml`.
+* *build-deps-pom(-sources)* + *coverage-deps-pom*
+    - used by the `ant build` target. Listed dependencies will be resolved and
+      copied to `build/lib/{jar,sources}` by executing the
+      `maven-ant-tasks-retrieve-build` target. This should contain libraries that are
+      required for build tools (grammar, docs, instrumentation), but are not
+      shipped as part of the Cassandra distribution.
+    - Since Cassandra 4.0, `coverage-deps-pom` has been removed and the
+      `build-deps-pom` template is `.build/cassandra-build-deps-template.xml`.
+* *all-pom*
+    - POM for https://mvnrepository.com/artifact/org.apache.cassandra/cassandra-all[cassandra-all.jar]
+      that can be installed or deployed to public maven repos via `ant publish`.
```

Comment:
no you can't.  nothing can be pushed public without a vote and three PMC +1s

(what `ant publish` does it stage it the ASF nexus, where we can then vote on it.  but no one AFAIK needs to do this because we only do it as part of our release process.)

when a vote is finalised the staging repo in ASF nexus is "published" and gets pushed out to other public repos.

Reply:
This was an existing part of the documentation, but I just made some clarifications - let me know if this is suitable for you